pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -603,6 +602,7 @@ func (gb *GraphBuilder) processGraphChanges() bool {
 			existingNode.identity = observedIdentity
 			gb.uidToNode.Write(existingNode)
 		}
+		existingNode.markObserved()
```

Comment:
moved markObserved after cloning/replacing, so we don't mark the virtual node that disagreed with our coordinates as observed